conclave-host/src/main/kotlin/com/r3/conclave/host/internal/SocketHostEnclaveInterface.kt

```diff
@@ -55,19 +57,42 @@ class SocketHostEnclaveInterface : CallInterface<EnclaveCallType, HostCallType>(
         }
     }
 
-    /** Start the call interface and allow calls to begin. */
-    fun start() {
+    /**
+     * Start the call interface and allow calls to begin.
+     * Will poll "everythingOkay".
+     */
+    fun start(everythingOkay: () -> Boolean = {true}) {
         synchronized(stateManager) {
             if (stateManager.state == State.Running) return
             stateManager.transitionStateFrom<State.Ready>(to = State.Running)
 
             try {
                 /** Wait for IO sockets, then close the server socket. */
                 serverSocket.use {
+                    /**
+                     * Attempt to receive the maximum number of concurrent calls from the enclave.
+                     * Check the enclave subprocess periodically to ensure that it's still alive.
+                     */
+                    var connectionSuccessful = false
 
-                    /** Receive the maximum number of concurrent calls from the enclave. */
-                    it.accept().use { initialSocket ->
-                        maxConcurrentCalls = DataInputStream(initialSocket.getInputStream()).readInt()
+                    val initialConnectionThread = Thread {
+                        it.accept().use { initialSocket ->
```

Comment:
This `it` is quite removed from where it's defined. perhaps give it a name to make it clearer.